src/app/report/report-keywords.component.ts

```diff
@@ -0,0 +1,48 @@
+import {Component, Input} from 'angular2/core';
+
+@Component({
+  selector: 'report-keywords',
+  styleUrls: ['app/report/report-keywords.component.css'],
+  template: `
+    <section>
+      <h4>This episode's unique identifying keyword:</h4>
+      <div class="id-keyword">
+        <input type="text" readonly [value]="idKeyword" #idKeywordInput>
+        <button
+          (mouseover)="resetCopyButton(copyButton)"
+          (click)="copyCode(idKeywordInput, copyButton)"
+          #copyButton>
+          Copy
```

Comment:
I might've just used the framework's interpolation here rather than setting innerHTML explicitly, i.e. {{ buttonText }}, but this works fine. Plus you're already working directly with HTML elements, so it's fitting.